### PR TITLE
Fix dev dump and restore scripts

### PR DESCRIPTION
**Pull request by bion**
The dump script appeared to work but actually generates a corrupted binary archive, I think because docker might be doing something to stdout output. This change fixes that by mounting the project directory into the docker image and writing the archive file in there instead of from the output from docker.

The line continuation in the restore script was wrong.

### CODE REVIEW

File: bin/dev-restore-db [hunk 1]
```diff
@@ -33,8 +33,9 @@ readonly DUMP_FILE="project/${1}"
 readonly TRUNCATE_DB_CMD="/usr/bin/psql -h db -U postgres -c '${TRUNCATE_SQL}' postgres"
 
 # command for invoking pg_restore with the dev database
-readonly PG_RESTORE_CMD="/usr/bin/pg_restore -h db -U postgres -d postgres"\
-  " < ${DUMP_FILE}"
+
+readonly PG_RESTORE_CMD="/usr/bin/pg_restore -h db -U \
+  postgres -d postgres < ${DUMP_FILE}"
```
Comment (zaralouis-sf) [inline bin/dev-restore-db:38]: These file changes are just formatting right? Want to make sure I'm not missing anything

Comment (bion) [reply bin/dev-restore-db:38]: This actually changes behavior somehow. I'm still fuzzy on how line continuations work in strings with bash but I tested this script with the output of the other one and it gives a convincing appearance of working.

File: bin/dump-dev-program-config [hunk 2]
```diff
@@ -7,17 +7,18 @@ source bin/lib.sh
...
+    -w \
+    -Fc \
+    -h db \
+    -U postgres \
+    -t programs \
+    -t questions \
+    -t versions \
+    -t versions_programs \
+    -t versions_questions \
+    postgres > /project/dev_db.dump"
```
Comment (zaralouis-sf) [inline bin/dump-dev-program-config:24]: Nit: I think you need to update this name in the docs at the top of this file

Comment (bion) [reply bin/dump-dev-program-config]: aha, reverted to the original name. thanks!

File: bin/dump-dev-program-config [hunk 3]
```diff
@@ -7,17 +7,18 @@ source bin/lib.sh
...
-  -Fc \
-  -h db \
-  -U postgres \
-  -t programs \
-  -t questions \
-  -t versions \
-  -t versions_programs \
-  -t versions_questions \
-  postgres > dev_programs.dump
+  /bin/bash -c "pg_dump \
```
Comment (zaralouis-sf) [inline bin/dump-dev-program-config:14]: Why was this changed from /usr/bin? And why does the other file still use usr/bin ?

Comment (bion) [reply bin/dump-dev-program-config:14]: tbh mostly habit and tradition, at least for bash.

system binaries are mostly in `/bin` whereas user binaries often go in `/usr/bin`. it's common to invoke the system version of bash and afaik on most linux distros `/bin/bash` and `/usr/bin/bash` are actually the same inode.

in the case of `pg_dump`, it's userland program rather than a system utility, so it's unlikely to ever be in `/bin`

**Review by zaralouis-sf — APPROVED** [review]
